test/SourceKit/CodeComplete/complete_override.swift.response

```diff
@@ -7,6 +7,7 @@
       key.description: "associatedtype",
       key.typename: "",
       key.context: source.codecompletion.context.none,
+      key.typerelation: source.codecompletion.typerelation.unrelated,
```

Comment:
Keyword completions that have no type should not have a type relation.  Exceptions: `nil`, `true`, `false`, etc.

Reply:
We could have 6 state for this.

* not available: The result is not relevant for type relation (e.g. keywords, overloads)
* unknown: We didn't calculate the relation (cached results), or we don't know the context type.
* invalid: The result type is invalid for this context (i.e. `Void` for non-`Void` context)
* unrelated: The result type has no relation to the context type
* convertible: The result type is convertible to the context type
* identical: The result type is identical to the context type

Reply:
Good point. I would call it "not **applicable**" rather than "not available" to emphasize that the difference between this case and "unknown" is that we know for sure it makes no sense.